src/Ubiquity/db/providers/pdo/drivers/PgsqlDriverMetas.php

```diff
@@ -13,19 +13,86 @@
 class PgsqlDriverMetas extends AbstractDriverMetaDatas {
 
 	public function getForeignKeys($tableName, $pkName, $dbName = null): array {
-		// TODO
+		$recordset = $this->dbInstance->query ( '
+			SELECT k1.constraint_catalog as "CONSTRAINT_CATALOG",
+			k1.constraint_schema as "CONSTRAINT_SCHEMA",
+			k1.constraint_name as "CONSTRAINT_NAME",
+			k1.table_catalog  as "TABLE_CATALOG",
+			k1.table_schema  as "TABLE_SCHEMA",
+	        k1.table_name  as "TABLE_NAME",
+	        k1.column_name as "COLUMN_NAME",
+	  	    k1.ordinal_position as "ORDINAL_POSITION" ,
+	  	    k1.position_in_unique_constraint as "POSITION_IN_UNIQUE_CONSTRAINT",
+	        k2.table_schema AS "REFERENCED_TABLE_SCHEMA",
+	        k2.table_name AS "REFERENCED_TABLE_NAME",
+	        k2.column_name AS "REFERENCED_COLUMN_NAME"
+			FROM information_schema.key_column_usage k1
+			JOIN information_schema.referential_constraints fk USING (constraint_schema, constraint_name)
+			JOIN information_schema.key_column_usage k2
+			  ON k2.constraint_schema = fk.unique_constraint_schema
+			AND k2.constraint_name = fk.unique_constraint_name
+			AND k2.ordinal_position = k1.position_in_unique_constraint
+		 		WHERE k1.table_schema = \'public\'
+		 		and k2.column_name=\'' . $pkName . '\'
+		   		AND k1.table_name   = \'' . $tableName . '\';' );
+		return $recordset->fetchAll ( \PDO::FETCH_ASSOC );		
 	}
 
 	public function getTablesName(): array {
-		// TODO
+		$query = $this->dbInstance->query ( 'SELECT tablename as schemaname FROM pg_catalog.pg_tables WHERE schemaname != \'pg_catalog\' AND schemaname != \'information_schema\';'  );
+		return $query->fetchAll ( \PDO::FETCH_COLUMN );
 	}
 
 	public function getPrimaryKeys($tableName): array {
-		// TODO
+		$fieldkeys = array ();
+		$recordset = $this->dbInstance->query ( "SELECT a.attname, format_type(a.atttypid, a.atttypmod) AS data_type FROM   pg_index i JOIN   pg_attribute a ON a.attrelid = i.indrelid AND a.attnum = ANY(i.indkey) WHERE i.indrelid = '{$tableName}'::regclass AND    i.indisprimary;" );
+		$keys = $recordset->fetchAll ( \PDO::FETCH_ASSOC );
+		foreach ( $keys as $key ) {
+			$fieldkeys [] = $key ['attname'];
+		}
+		return $fieldkeys;
 	}
 
 	public function getFieldsInfos($tableName): array {
-		// TODO
+		$fieldsInfos = array ();
+		$recordset = $this->dbInstance->query ( 
+					"SELECT  
+			f.attname AS \"Field\",  
+			pg_catalog.format_type(f.atttypid,f.atttypmod) AS \"Type\",
+			CASE  
+			    WHEN f.attnotnull=true THEN 'YES' 
+			    WHEN f.attnotnull=false THEN 'NO' 
+			    ELSE ''
+			END AS \"Null\",
+			CASE  
+			    WHEN p.contype = 'u' THEN 'MUL' 
+			    WHEN p.contype = 'p' THEN 'PRI' 
+			    ELSE ''
+			END AS \"Key\",
+			CASE
+			    WHEN f.atthasdef = 't' THEN d.adsrc
+			END AS \"Default\"  ,
+			CASE WHEN d.adsrc LIKE 'nextval(%' THEN 'auto_increment' ELSE '' END AS \"Extra\" 
```

Comment:
the `adsrc` field is no longer present in the `pg_attrdef` table in version 12 of **pgsql**.
Since version 8, it was maintained for historical reasons.
see [v11](https://www.postgresql.org/docs/11/catalog-pg-attrdef.html) and [v12](https://www.postgresql.org/docs/12/catalog-pg-attrdef.html)

So an alternative should be found:
As the pgsql documentation says
>Reverse-compiling the adbin field (with pg_get_expr for example) is a better way to display the default value

Reply:
And it's mostly that we only need the `Type` and the `Nullable`!

Reply:
i updated both lines. review latest one.
I also Added SQLite PDo Driver. It connects but i couldnt test from models generation database.

Reply:
"And it's mostly that we only need the Type and the Nullable!"  i couldnt be sure about it.
In order to be compatible , I created all fields same as mysql order. I thought maybe for future it can be beneficial when it is compatible with mysql. because, all query things in framework based on mysql.

Reply:
Can you do another PR for SQLite,
we'll do the integration in 2 steps (1 postgres , 2 sqlite)

Reply:
> "And it's mostly that we only need the Type and the Nullable!" i couldnt be sure about it because i couldnt read all code.
> In order to be compatible , I created all fields same as mysql order. I thought maybe for future, it can be beneficial when it is compatible with mysql. because, all query things in framework based on mysql.

Yes, we can leave the full query.
And this part has no performance implications.

Reply:
I coulndt make another PR because i cant see the way.I created branch but it think i have to commit again the files related to SQLite. sorry, it is first time for me.

Reply:
You create a new `add-sqlite-driver` branch based on the original branch of your fork.
there you modify the `SqliteDriverMeta` class

Then you can do new PR.

On your `add-pdo-driver-metas` branch (this PR), you delete your Sqlite-related changes, and you commit again.

Reply:
I reverted the SQlite and PostgreSQL commits. and only postgreSQL commits left.
Now i will try to Create New PR.

Reply:
I was going to tell you it's okay, forget it.
but since you cleaned up...

Reply:
I reverted and again commited only SQLite but this time again it adds to this PR. not New PR button shows up.
Sorry about that. you may review only PostgreSQL for this. maybe after you accepted this PR I can Create new PR.